webpack.config.js

```diff
@@ -67,7 +71,9 @@ function generateWebpackConfigForCanister(name, info) {
        { test: /\.css$/, use: ['style-loader','css-loader'] }
      ]
     },
-    plugins: [],
+    plugins: [
+      new webpack.EnvironmentPlugin(['CANISTER_ID_whoami']),
```

Comment:
This won't work until next release of `dfx` :/

There's another var tho (prefixed with `CANISTER_`) that is the path to a file that contains the canister id, and you'd have to use node `fs` to parse it out of that file.

Reply:
It works now with `dfx@latest` (which is currently `0.6.25`)